urwid/display_common.py

```diff
@@ -1004,7 +1003,7 @@ class BaseScreen(metaclass=BaseMeta):
     def __init__(self) -> None:
         super().__init__()
 
-        self.logger = LOGGER.getChild(self.__class__.__name__)
```

Comment:
this logger could be part of the class instead of the instance so it doesn't have to be tracked in every single widget

Reply:
It will require metaclass usage, which is a bit overkill

Reply:
It could also be a Widget property or method that returns the logger

Reply:
Technically once logger initializer, it stored in `logging.Manager`, so all instances of the same `Widget` class uses the same `Logger`.
Adding property may break downstream, because it will cause errors if downstream initialize `logger` itself.